### ISSUE TITLE
Avatar removal is not detected

### ISSUE THREAD

**Issue by MilanKral** [cmeng-git/atalk-android#143]
When a contact updates his/her avatar this change is correctly processed in aTalk.
But when a contact removes his/her avatar without replacing it with a new avatar aTalk will still display the old avatar.

**Comment by cmeng-git** [comment]
May I want which client did you test with aTalk. Also do you know what is the protocol that the client is supported i.e.

XEP-0084: User Avatar
XEP-0153: vCard-Based Avatar

**Comment by cmeng-git** [comment]
Per the [XEP-0084: User Avatar](https://xmpp.org/extensions/xep-0084.html)

section 3.5 Publisher Disables Avatar Publishing:
The document states that user can temporarily "Disables Avatar Publishing", however there is no mention that subscribers of the user need to take any action to remove/clear the user avatar.

Also in [XEP-0153: vCard-Based Avatars](https://xmpp.org/extensions/xep-0153.html)
user can remove the avatar from the server, however there is no requirement for the other contacts to remove the avatar when this happen.  

In conclusion, what you have observed on aTalk is the correct behavior i.e. aTalk does not clear a contact avatar when the contact  temporarily "Disables Avatar Publishing"  or remove his/her avatar info from his vcard info.

**Comment by MilanKral** [comment]
ok, so this is not an error. I'm using Gajim for testing.

I have another question. When long press is done on the avatar icon for a client, aTalk will display message "Retrieving avatar for ...".
But when there is already an avatar icon for the contact no XMPP request is send to the server.
Only when there is none avatar icon for the contact a XMPP request is send to the server.

**Comment by cmeng-git** [comment]
This is aTalk implementation to overcome some shortfall when a client supports only 
XEP-0153: vCard-Based Avatar protocol.

In this protocol, the server only sends on behalf of contact, the avatar photo-hash info when he/she is online (see below). When a subscribed user receives this \<presence/> info, aTalk will check the received photo-hash against the local stored avatar hash, and proceed to fetch vCard info from the server if it cannot find a corresponding avatar (hash) or the received photo-hash is different from the local stored avatar hash. Otherwise aTalk will just use the local stored avatar for display.

When a new user is first registered with the server, he will receive all the buddy jids of the shared-roster. The user client will then starts to populate the buddies' avatars only for all the online buddies i.e. \<presence> received; but show a blank avatar for those contacts whom are not online. The user can continue to wait for the missing online buddies to go online to update his/her avatar. In aTalk, user can manually download the buddy avatar by long click on the avatar icon. aTalk will check for local available avatar in store. When no found, it will proceed to fetch the avatar (VCard), keep a local copy and display the fetch avatar if any; and display "Retrieving avatar for ...". 

```
    <presence xml:lang='en' to='swordfish@atalk.sytes.net/atalk' from='xyz123@atalk.sytes.net/Conversations.CPqI'>
      <c xmlns='http://jabber.org/protocol/caps' ver='PeKYibv9P9qF24DQsRGfza05Bn0=' node='http://conversations.im' hash='sha-1'/>
      <x xmlns='vcard-temp:x:update'>
        <photo>
          050be6d940f8c5515bafcc95f02f55a87615ffad
        </photo>
      </x>
      <delay from='xyz123@atalk.sytes.net/Conversations.CPqI' stamp='2020-04-27T08:37:16.963290Z' xmlns='urn:xmpp:delay'/>
    </presence>
```
